cve_bin_tool/cvedb.py

```diff
@@ -787,15 +842,61 @@ def json_to_db_wrapper(self, path):
             metadata_fd = open(path / "metadata.json")
             metadata = json.loads(metadata_fd.read())
             metadata_fd.close()
+            if not metadata["signed"]:
+                LOGGER.warning(
+                    "Importing JSON data that is not signed, the JSON data might have been tampared with"
+                )
+            elif not pubkey and not ignore_signature:
+                LOGGER.critical(
+                    "The JSON data is signed, so you must provide the public key with --verify flag or use --ignore-sig flag to skip signature verification"
+                )
+                return ERROR_CODES[SigningError]
+
+            if metadata["signed"] and not ignore_signature:
+                temp_gnupg_home = Path(tempfile.mkdtemp(prefix=".gnupg-"))
+                gpg = gnupg.GPG(gnupghome=temp_gnupg_home)
+                key_import_status = gpg.import_keys_file(pubkey)
+                if key_import_status.results[0]["fingerprint"] is None:
+                    LOGGER.critical(
+                        "Something went wrong while importing the public key. Please try again!"
+                    )
+                    shutil.rmtree(temp_gnupg_home)
+                    return ERROR_CODES[SigningError]
+
+            if metadata["signed"] and not ignore_signature:
+                with open(path / "metadata.json", "rb") as fd:
+                    is_verified = gpg.verify_data(path / "metadata.asc", fd.read())
+                if not is_verified:
+                    LOGGER.critical("Invalid signature detected!")
+                    shutil.rmtree(temp_gnupg_home)
+                    return ERROR_CODES[SigningError]
+
             for dir in track(
                 metadata["db"], description="Import database from JSON files...."
             ):
                 for year in (path / dir).iterdir():
+                    if not str(year).endswith(".json"):
+                        continue
                     json_fd = open(year)
-                    data = json.loads(json_fd.read())
+                    data = json_fd.read()
                     json_fd.close()
-                    self.json_to_db(cursor, dir, data)
+                    if metadata["signed"] and not ignore_signature:
+                        signature_path = str(
+                            path / dir / str(year).replace(".json", ".asc")
+                        )
+                        is_verified = gpg.verify_data(
+                            signature_path, data.encode("utf-8")
+                        )
+                        if not is_verified:
+                            LOGGER.critical("Invalid signature detected!")
+                            shutil.rmtree(temp_gnupg_home)
+                            return ERROR_CODES[SigningError]
+                    self.json_to_db(cursor, dir, json.loads(data))
                     self.connection.commit()
+
+            if metadata["signed"] and not ignore_signature:
```

Comment:
Needs a change, maybe a check for the existence of `temp_gnupg_home` in this case.